src/node/chainstate.h

```diff
@@ -58,32 +54,11 @@ enum class ChainstateLoadingError {
  *  4. Hard failure
  *    - a failure that definitively cannot be recovered from with a reindex
  *
- *  Currently, LoadChainstate returns a std::optional<ChainstateLoadingError>
- *  which:
- *
- *  - if has_value()
- *      - Either "Soft failure", "Hard failure", or "Shutdown requested",
- *        differentiable by the specific enumerator.
- *
- *        Note that a return value of SHUTDOWN_PROBED means ONLY that "during
- *        this sequence, when we explicitly checked shutdown_requested() at
- *        arbitrary points, one of those calls returned true". Therefore, a
- *        return value other than SHUTDOWN_PROBED does not guarantee that
- *        shutdown hasn't been called indirectly.
- *  - else
- *      - Success!
+ *  LoadChainstate returns a (status code, error string) tuple.
  */
-std::optional<ChainstateLoadingError> LoadChainstate(ChainstateManager& chainman, const CacheSizes& cache_sizes,
-                                                     const ChainstateLoadOptions& options);
-
-enum class ChainstateLoadVerifyError {
-    ERROR_BLOCK_FROM_FUTURE,
-    ERROR_CORRUPTED_BLOCK_DB,
-    ERROR_GENERIC_FAILURE,
-};
-
-std::optional<ChainstateLoadVerifyError> VerifyLoadedChainstate(ChainstateManager& chainman,
-                                                                const ChainstateLoadOptions& options);
+ChainstateLoadResult LoadChainstate(ChainstateManager& chainman, const CacheSizes& cache_sizes,
+                                    const ChainstateLoadOptions& options);
```

Comment:
second commit: I wonder if it was easier to review if the two functions were converted in separate commits?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/25308#discussion_r919898257

> second commit: I wonder if it was easier to review if the two functions were converted in separate commits?

I think it would be harder to review and introduce the potential for bugs in the intermediate commit. If you look code calling these functions in init.cpp you can see it has a lot of cases that need to be handled, and a twisty control flow involving exceptions and changing intermediate variables and timing and log statements. It seems like less of a burden on reviewers and less error prone to update the init.cpp code in one shot, than to change the same code twice and leave it in an intermediate state where the load and verify calls don't have consistent return styles.